extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java

```diff
@@ -407,15 +411,35 @@ public boolean hasNext() {
         public BindingSet next() {
             final MapBindingSet bs = new MapBindingSet();
 
-            for(Binding binding : newResult) {
+            for(final Binding binding : newResult) {
                 bs.addBinding(binding);
             }
 
-            for(Binding binding : joinedResults.next()) {
+            final BindingSet joinResult = joinedResults.next();
+            for(final Binding binding : joinResult) {
                 bs.addBinding(binding);
             }
 
-            return bs;
+            String visibility = "";
+            if(newResult instanceof VisibilityBindingSet && joinResult instanceof VisibilityBindingSet) {
```

Comment:
This is another area where we should be able to assume we're using VisibilityBindingSets.